src/App.js

```diff
@@ -37,14 +41,54 @@ class App extends Component {
       })
   }
 
+  selectGem = (id) => {
+   let selectedGem = this.state.gems.find((gem) => {
+      return gem.id === id
+    })
+    this.setState({
+      selectedGem
+    })
+    this.changeDisplayPage('metals')
+  }
+
+  selectMetal = (id) => {
+    let selectedMetal = this.state.metals.find((metal) => {
+      return metal.id === id
+    })
+    this.setState({
+      selectedMetal
+    })
+    this.changeDisplayPage('results')
+  }
+
+  changeDisplayPage = (nextPage) => {
+    this.setState({
+      display: nextPage
+    })
+  }
+
   render() {
     return (
       <div>
-        <SplashPage />
-        <GemsPage gems={this.state.gems} metals={this.state.metals}/>
+        {
+          this.state.display === 'gems' &&
+          <GemsPage selectGem={this.selectGem} gems={this.state.gems}/>
+        }
+        {
+          this.state.display === 'metals' &&
+          <MetalsPage selectMetal={this.selectMetal} metals={this.state.metals}/>
+        }
+        {
+          this.state.display === 'results' &&
+          <ResultsPage />
```

Comment:
In general, you want to avoid having state properties that represent what components should/shouldn't be shown. e.g. `display` changes which components are shown simply because it says so --- but what does it really mean for each of these different components to be rendered? If someone has selected a gem, we should render the `Gems` component. So our conditional logic here should really be something like:

`{ this.state.selectedGem && <Gems /> }`

This can be tricky when your default value for the state is an object, since it would always return true. So you might instead need to check for something more specific -- like `this.state.gem.name` to determine if that object actually has properties on it or not.